tests/test_model.py

```diff
@@ -0,0 +1,234 @@
+            circuit()
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.model.Interferometer(U=np.identity(self.num_subsystems), phi=np.zeros(int(self.num_subsystems*(self.num_subsystems-1)/2)), wires=range(self.num_subsystems))
+            return qml.expval.X(wires=0)
+
+        with self.assertRaises(ValueError):
+            circuit()
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.model.Interferometer(phi=np.zeros(int(self.num_subsystems*(self.num_subsystems-1)/2)), wires=range(self.num_subsystems))
+            return qml.expval.X(wires=0)
+
+        with self.assertRaises(ValueError):
+            circuit()
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.model.Interferometer(theta=np.zeros(int(self.num_subsystems*(self.num_subsystems-1)/2)), wires=range(self.num_subsystems))
+            return qml.expval.X(wires=0)
+
+        with self.assertRaises(ValueError):
+            circuit()
+
+    def test_clements_non_square_error(self):
+        V_non_square = np.random.randn(2, 3)
+        with self.assertRaises(ValueError):
+            qml.model._clements(V_non_square)
+
+class TestCVNeuralNet(BaseTest):
+    """Tests for the CVNeuralNet and CVNeuralNetLayer from the pennylane.model module."""
+
+    def setUp(self):
+        super().setUp()
+        np.random.seed(8)
+        self.num_wires = 4
+
+    def test_cvqnn(self):
+
+        dev = qml.device('default.gaussian', wires=self.num_wires)
+
+        # weights = np.array([np.array([
+        #     # np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)),
+        #     # np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)),
+        #     # np.random.uniform(0.1, 0.7, self.num_wires),
+        #     # np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)),
+        #     np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)),
+        #     np.random.uniform(0.1, 2.0, self.num_wires),
+        #     np.random.uniform(0.1, 0.3, self.num_wires)
+        # ]) for l in range(2)])
+
+        weights = [[
+            np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #transmittivity angles
+            np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #phase angles
+            np.random.uniform(0.1, 0.7, self.num_wires), #squeezing amounts
+            np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #transmittivity angles
+            np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #phase angles
+            np.random.uniform(0.1, 2.0, self.num_wires), #displacement magnitudes
+            np.random.uniform(0.1, 0.3, self.num_wires) #kerr parameters
+        ] for l in range(2)]
+
+        with patch.object(Kerr, '__init__', return_value=None) as _: #todo: Kerr() does not work on any core device, so we have to mock it here with a trivial class
```

Comment:
Here I have to "mock away" the Kerr Operations, as they are not supported by any core PL device.